larpandoracontent/LArControlFlow/TrackDirectionTool.h

```diff
@@ -0,0 +1,1089 @@
+{
+    return m_uncertainty;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline void TrackDirectionTool::HitCharge::SetDistanceToNN(float &distance)
+{
+    m_distancetonearestneighbour = distance;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float TrackDirectionTool::HitCharge::GetDistanceToNN() const
+{
+    return m_distancetonearestneighbour;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+inline void TrackDirectionTool::HitCharge::SetForwardsFitCharge(float &Q_fit_f) 
+{
+    m_forwardsfitcharge = Q_fit_f;
+}
+//------------------------------------------------------------------------------------------------------------------------------------------
+inline void TrackDirectionTool::HitCharge::SetForwardsSigma(float &f_sigma)
+{
+    m_forwardssigma = f_sigma;
+}
+//------------------------------------------------------------------------------------------------------------------------------------------
+inline void TrackDirectionTool::HitCharge::SetForwardsDelta(float &forwardsDelta)
+{
+    m_forwardsdelta = forwardsDelta;
+}
+//------------------------------------------------------------------------------------------------------------------------------------------
+inline void TrackDirectionTool::HitCharge::SetForwardsChiSquared(float &forwardsHitChisquared)
+{
+    m_forwardschisquared = forwardsHitChisquared;
+}
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline void TrackDirectionTool::HitCharge::SetBackwardsFitCharge(float &Q_fit_b) 
+{
+    m_backwardsfitcharge = Q_fit_b;
+}
+//------------------------------------------------------------------------------------------------------------------------------------------
+inline void TrackDirectionTool::HitCharge::SetBackwardsSigma(float &b_sigma)
+{
+    m_backwardssigma= b_sigma;
+}
+//------------------------------------------------------------------------------------------------------------------------------------------
+inline void TrackDirectionTool::HitCharge::SetBackwardsDelta(float &backwardsDelta)
+{
+    m_backwardsdelta = backwardsDelta;
+}
+//------------------------------------------------------------------------------------------------------------------------------------------
+inline void TrackDirectionTool::HitCharge::SetBackwardsChiSquared(float &backwardsHitChisquared)
+{
+    m_backwardschisquared = backwardsHitChisquared;
+}
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float TrackDirectionTool::HitCharge::GetForwardsFitCharge() 
```

Comment:
Can they? If I set those, and thus the public floats, and then further the relevant member variables to const I see a range of errors, mostly "non-static const member ‘const float lar_content::TrackDirectionTool::HitCharge::m_forwardsFitCharge’, can’t use default assignment operator" and "type qualifiers ignored on function return type [-Werror=ignored-qualifiers]
 inline const float TrackDirectionTool::HitCharge::GetForwardsFitCharge()"

Reply:
You definitely won't want to set the member variables to const; declaring a function to be const, i.e.

`inline float TrackDirectionTool::HitCharge::GetForwardsFitCharge() const`

is an agreement that the function being called doesn't do anything to change the values of member variables of the object against which the function is being called, rather than a statement about whether the values involved are themselves constants. So in this case, as far as I can see, all of these functions are just returning the current value of some variable, and so the function itself can be declared const (at which point, any attempt to edit a member variable in one of these functions would result in a compiler error).

Reply:
Ah okay. Done.